src/Microsoft.AspNetCore.Server.Kestrel.Core/Transport/IConnectionContext.cs

```diff
@@ -12,6 +12,7 @@ public interface IConnectionContext
         string ConnectionId { get; }
         IPipeWriter Input { get; }
         IPipeReader Output { get; }
+        void OnConnectionClosed();
```

Comment:
Nit: Move below "// TODO: Remove these (Use Pipes instead?)"

There is a task that will be added to Pipes that should let us track when SocketOutputConsumer complete's it's pipe reader which will be right after disposing the socket.